extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -590,10 +587,18 @@ export class Bind {
           // TODO(choumx): Perform in worker with URL API.
           // Rewrite attribute value if necessary. This is not done in the
           // worker since it relies on `url#parseUrl`, which uses DOM APIs.
-          const rewrittenNewValue = rewriteAttributeValue(
-              element.tagName, property, String(newValue));
-          element.setAttribute(property, rewrittenNewValue);
-          attributeChanged = true;
+          let rewrittenNewValue;
+          try {
+            rewrittenNewValue = rewriteAttributeValue(
+                element.tagName, property, String(newValue));
+          } catch (e) {
+            reportError(user().createError(e), element);
+          }
+          // Rewriting can fail due to e.g. invalid URL.
+          if (rewrittenNewValue !== undefined) {
+            element.setAttribute(property, rewrittenNewValue);
```

Comment:
That will harm readability since it'll be less clear what is throwing the error. Generally you should keep `try` blocks as small as possible.